packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1497,6 +1499,15 @@ class Navigator extends StatefulWidget {
   /// will be primed.
   final RouteListFactory onGenerateInitialRoutes;
 
+  /// Whether this navigator should report route update message back to the
+  /// engine when the top-most route changes.
+  ///
+  /// If the property is set to true, this navigator automatically sends the
+  /// route update message to the engine when it detects top-most route changes.
```

Comment:
Can you document the problems that this can cause if this is turned on for nested navigators (or whenever you have more than one navigator doing this)?